client/www/js/controllers/routeFilterController.js

```diff
@@ -26,7 +26,16 @@ angular.module('amblr.routeFilter', [])
 
     $scope.setRoute = function (routeID) {
       $scope.filteredRoute = routeID;
-      POIs.setRouteFilter(routeID);
+      if (routeID) {
```

Comment:
Needed because we pass routeID = null when we select show all POIs (no route selected).